client/src/fuzzy-search.js

```diff
@@ -33,18 +25,11 @@ const getSubstrings = (regexString, matchString) => {
   let matchArray = regex.exec(matchString);
   let substrings = [];
   let isMatch = false;
-  let missingCount = 0;
-  let startsWith = 0;
   for (let i = 1; i < matchArray.length; i++) {
     if (matchArray[i]) substrings.push({ str: matchArray[i], match: isMatch });
-    // if undefined and not first or last, add a point because it's a match combo
-    if (!matchArray[i] && i !== 1 && i !== matchArray.length - 1)
-      missingCount++;
-    // if first is undefined, string starts with match
-    if (!matchArray[i] && i === 1) startsWith++;
     isMatch = !isMatch;
```

Comment:
I suppose we only take every other match because of the capturing groups in the regex? Imo that deserves a more explicit var name a la `isCharMatch` or a comment.